### PR TITLE
fix: Add limit to number of Workflows in CronWorkflow history

### PR DESCRIPTION
**Pull request by panicboat**
Fixes https://github.com/argoproj/argo-workflows/issues/12645

### Motivation

I want to avoid argo-server going OOM when there is a lot of execution history.

### Modifications

Limited the number of histories to 50.
image
image

### Verification

- [x] No completed cron workflows
- [x] Cron workflow with the number of cases exceeding the limit
- [x] Cron workflow under limit

### CODE REVIEW

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 1]
```diff
@@ -21,12 +23,21 @@ import {CronWorkflowEditor} from '../cron-workflow-editor';
...
     const queryParams = new URLSearchParams(location.search);
 
     const [namespace] = useState(match.params.namespace);
+    const [pagination] = useState<Pagination>(() => {
+        const savedPaginationLimit = storage.getItem('options', {}).paginationLimit || undefined;
+        return {
+            offset: queryParams.get('offset') || undefined,
+            limit: parseLimit(queryParams.get('limit')) || savedPaginationLimit || 50
+        };
+    });
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:40]: I see that you copied this from [the code](https://github.com/argoproj/argo-workflows/blob/986b069f34ed5336b6f754cfc6709c30b302c768/ui/src/app/workflows/components/workflows-list/workflows-list.tsx#L56) in the Workflows List, however this does not make sense here.
This is currently creating `state` that is not used -- meaning it isn't stateful to begin with.

I notice that `namespace` and `name` in this function aren't stateful either, those should be converted to plain `const`s as well.

More specifically, this code allows the initial pagination to be set via `localStorage` and query params, but there is no way to change it after. In the Workflows List page, there is the `PaginationPanel` that can change those.

We also don't want it to be changeable -- it should be a hard-coded `50`

```suggestion
```

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 2]
```diff
@@ -21,12 +23,21 @@ import {CronWorkflowEditor} from '../cron-workflow-editor';
 import '../../../workflows/components/workflow-details/workflow-details.scss';
 import './cron-workflow-details.scss';
 
+const storage = new ScopedLocalStorage('ListOptions');
+
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:27]:
```suggestion
```

same as above, this is not needed

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 3]
```diff
@@ -8,6 +8,8 @@ import {uiUrl} from '../../../shared/base';
 import {ErrorNotice} from '../../../shared/components/error-notice';
 import {openLinkWithKey} from '../../../shared/components/links';
 import {Loading} from '../../../shared/components/loading';
+import {Pagination, parseLimit} from '../../../shared/pagination';
+import {ScopedLocalStorage} from '../../../shared/scoped-local-storage';
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:12]:
```suggestion
```

same as above, not needed

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 4]
```diff
@@ -71,7 +82,7 @@ export function CronWorkflowDetails({match, location, history}: RouteComponentPr
 
     useEffect(() => {
         (async () => {
-            const workflowList = await services.workflows.list(namespace, null, [`${models.labels.cronWorkflow}=${name}`], null);
+            const workflowList = await services.workflows.list(namespace, null, [`${models.labels.cronWorkflow}=${name}`], pagination);
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:85]:
```suggestion
            const workflowList = await services.workflows.list(namespace, null, [`${models.labels.cronWorkflow}=${name}`], {limit: 50});
```

this should just be hard-coded here

**Review by agilgur5 — CHANGES_REQUESTED** [review]
Some unnecessary and incorrect code was added here with regard to `state`, `localStorage`, and query params. See in-line comments below:

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 1 above)
Comment (panicboat) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx]: I see, thanks.
Now I understand why you did not add the pagination feature to this page.
I may issue an additional PR in the future when I get a chance, but for now, please let this be the response.

Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx]: A PR to add a link at the bottom would be good!

**Review by agilgur5 — APPROVED** [review]
Confirmed this is working as intended. Thanks for the fix!

**Comment by agilgur5** [comment]
Fixed merge conflicts with the added then removed imports.

**Pull request merged**